test/e2e/daemonset/daemonset.go

```diff
@@ -0,0 +1,39 @@
+package daemonset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	testclient "github.com/openshift/ingress-node-firewall/test/e2e/client"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func WaitForDaemonSetReady(client *testclient.ClientSet, ds *appsv1.DaemonSet, namespace, name string, retryInterval, timeout time.Duration) error {
+	err := wait.PollImmediate(retryInterval, timeout, func() (done bool, err error) {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
+		defer cancel()
+		err = client.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, ds)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		if ds.Status.DesiredNumberScheduled == ds.Status.NumberReady {
+			return true, nil
+		} else {
+			return false, nil
+		}
+	})
+	if err != nil {
+		fmt.Printf("failed to wait for ds %s/%s to be ready: %v", namespace, name, err)
+		return err
```

Comment:
u can probably remove fmt.Printf and return fmt.Errorf() instead

Reply:
also why writing this while we have it already here
https://github.com/openshift/ingress-node-firewall/blob/master/pkg/status/status.go#L101 ?

Reply:
I didnt see that but it also doesnt suit my purpose of polling until ds is available. I dont think its good to add retry logic to the test code and would rather it buried in `WaitForDaemonSetReady`.